driver-async/src/main/com/mongodb/async/client/gridfs/GridFSBucket.java

```diff
@@ -0,0 +1,294 @@
+     * @return a new GridFSBucket instance with the different chunk size in bytes
+     */
+    GridFSBucket withChunkSizeBytes(int chunkSizeBytes);
+
+    /**
+     * Create a new GridFSBucket instance with a different read preference.
+     *
+     * @param readPreference the new {@link ReadPreference} for the database
+     * @return a new GridFSBucket instance with the different readPreference
+     */
+    GridFSBucket withReadPreference(ReadPreference readPreference);
+
+    /**
+     * Create a new GridFSBucket instance with a different write concern.
+     *
+     * @param writeConcern the new {@link WriteConcern} for the database
+     * @return a new GridFSBucket instance with the different writeConcern
+     */
+    GridFSBucket withWriteConcern(WriteConcern writeConcern);
+
+    /**
+     * Create a new MongoDatabase instance with a different read concern.
+     *
+     * @param readConcern the new {@link ReadConcern} for the database
+     * @return a new GridFSBucket instance with the different ReadConcern
+     * @mongodb.server.release 3.2
+     * @since 3.3
+     * @mongodb.driver.manual reference/readConcern/ Read Concern
+     */
+    GridFSBucket withReadConcern(ReadConcern readConcern);
+
+    /**
+     * Opens a Stream that the application can write the contents of the file to.
+     * <p>
+     * As the application writes the contents to the returned Stream, the contents are uploaded as chunks in the chunks collection. When
+     * the application signals it is done writing the contents of the file by calling close on the returned Stream, a files collection
+     * document is created in the files collection.
+     * </p>
+     *
+     * @param filename the filename for the stream
+     * @return the GridFSUploadStream that provides the ObjectId for the file to be uploaded and the Stream to which the
+     * application will write the contents.
+     */
+    GridFSUploadStream openUploadStream(String filename);
+
+    /**
+     * Opens a Stream that the application can write the contents of the file to.
+     * <p>
+     * As the application writes the contents to the returned Stream, the contents are uploaded as chunks in the chunks collection. When
+     * the application signals it is done writing the contents of the file by calling close on the returned Stream, a files collection
+     * document is created in the files collection.
+     * </p>
+     *
+     * @param filename the filename for the stream
+     * @param options  the GridFSUploadOptions
+     * @return the GridFSUploadStream that provides the ObjectId for the file to be uploaded and the Stream to which the
+     * application will write the contents.
+     */
+    GridFSUploadStream openUploadStream(String filename, GridFSUploadOptions options);
+
+    /**
+     * Uploads a user file to a GridFS bucket.
+     * <p>
+     * Reads the contents of the user file from the {@code Stream} and uploads it as chunks in the chunks collection. After all the
```

Comment:
"Stream" is not code, since it matches neither the interface name or the parameter name.